code/modules/organs/organ_external.dm

```diff
@@ -698,20 +705,58 @@ Note that amputating the affected organ does in fact remove the infection from t
 			spawn(10)
 				del(spark_system)
 
-		if(organ)
-			owner.visible_message("\red [owner.name]'s [display_name] flies off in an arc.",\
-			"<span class='moderate'><b>Your [display_name] goes flying off!</b></span>",\
-			"You hear a terrible sound of ripping tendons and flesh.")
+		switch(disintegrate)
+			if(1)
+				owner.visible_message(
+					"<span class='danger'>\The [owner]'s [display_name] flashes away into ashes!</span>",\
+					"<span class='moderate'><b>Your [display_name] flashes away into ashes!</b></span>",\
+					"<span class='danger'>You hear the crackling sound of burning flesh.</span>")
+				new /obj/effect/decal/cleanable/ash(get_turf(owner))
+			if(2)
+				owner.visible_message(
+					"<span class='danger'>\The [owner]'s [display_name] explodes in a shower of gore!</span>",\
+					"<span class='moderate'><b>Your [display_name] explodes in a shower of gore!</b></span>",\
+					"<span class='danger'>You hear the sickening splatter of gore.</span>")
+				var/obj/effect/decal/cleanable/blood/gibs/gore = new(get_turf(owner))
+				if(owner.species.flesh_color)
+					gore.fleshcolor = owner.species.flesh_color
+				if(owner.species.blood_color)
+					gore.basecolor = owner.species.blood_color
+				gore.update_icon()
+				gore.throw_at(get_edge_target_turf(owner,pick(alldirs)),rand(1,3),30)
+
+				if(body_part == HEAD && owner.brain_op_stage != 4.0)
+					var/mob/living/simple_animal/borer/borer = owner.has_brain_worms()
+					if(borer)
+						borer.detatch() //Should remove borer if the brain is removed - RR
+					if(owner.species && owner.species.flags & IS_SYNTHETIC)
+						var/obj/item/device/mmi/posibrain/P = new(owner.loc)
+						P.transfer_identity(owner)
+						if(istype(P.loc,/turf))
+							P.throw_at(get_edge_target_turf(owner,pick(alldirs)),rand(1,3),30)
+					else
+						var/obj/item/brain/B = new(owner.loc)
+						B.transfer_identity(owner)
+						if(istype(B.loc,/turf))
+							B.throw_at(get_edge_target_turf(owner,pick(alldirs)),rand(1,3),30)
+
 
+
+		if(organ)
+			if(!amputation)
```

Comment:
I'd still make amputations give the "flies off in an arc", but that's me; it's rather hilarious =p Not remotely important.